### ISSUE TITLE
`riscv`: All the CSR write operations should be unsafe by default

### ISSUE THREAD

**Issue by jsgf** [rust-embedded/riscv#209]
In general we should assume that writing to CSRs could do something that potentially violates the Rust abstract model.

Macros like `write_csr_as` and `write_csr_as_usize` should at least default to `unsafe`, and maybe have an option to make a safe variant on a CSR by CSR basis.

**Comment by romancardenas** [comment]
As I always do, I checked the [`cortex-m`]/(https://github.com/rust-embedded/cortex-m/blob/4395baee03d57a4e1763c427073da5f33ec3b17e/cortex-m/src/register/lr.rs#L18) crate and... they use the same fashion you propose 😁 

In other words, I am in favor of your proposal.

**Comment by alistair23** [comment]
I agree. Writing to RISC-V CSRs can cause all sorts of `unsafe` behaviour. There are a few that probably aren't `unsafe` but they need some reasoning about why they aren't. `unsafe` by default is the way to go
